### PR TITLE
[cli] Expose system envs when autoExposeSystemEnvs is enabled on vc dev

### PR DESCRIPTION
**Pull request by nkzawa**
Ref: https://app.clubhouse.io/vercel/story/15112

We added a property called `autoExposeSystemEnvs` to projects. If that property is `true`, we automatically expose system env variables such as `VERCEL=1`, `VERCEL_ENV=<production | preview>`, ... to the runtime and build time.

This PR makes sure we mirror this behavior when running `vc dev` locally.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/now-cli/src/util/dev/builder.ts [hunk 1]
```diff
@@ -64,6 +66,7 @@ async function createBuildProcess(
   let PATH = `${dirname(process.execPath)}${delimiter}${process.env.PATH}`;
 
   const env: Env = {
+    ...systemEnvs.buildEnv,
     ...process.env,
```
Comment (styfle) [inline packages/now-cli/src/util/dev/builder.ts:69]: Do we want the local process.env to overwrite system env vars? Seems like it should be the other way around.

File: packages/now-cli/src/util/dev/server.ts [hunk 2]
```diff
@@ -1997,6 +2011,7 @@ export default class DevServer {
       // Most frameworks use `chalk`/`supports-color` so we enable it anyway.
       FORCE_COLOR: process.stdout.isTTY ? '1' : '0',
       ...(this.frameworkSlug === 'create-react-app' ? { BROWSER: 'none' } : {}),
+      ...this.systemEnvs.runEnv,
       ...process.env,
```
Comment (styfle) [inline packages/now-cli/src/util/dev/server.ts:2014]: Same question here

File: packages/now-cli/test/integration.js [hunk 3]
```diff
@@ -762,6 +763,71 @@ test('Deploy `api-env` fixture and test `vercel env` command', async t => {
     t.is(homeJson['VERCEL_URL'], localhostNoProtocol);
     t.is(homeJson['MY_STDIN_VAR'], '{"expect":"quotes"}');
 
+    // system env vars are not automatically exposed
+    t.is(apiJson['VERCEL'], undefined);
+    t.is(homeJson['VERCEL'], undefined);
+
+    vc.kill('SIGTERM', { forceKillAfterTimeout: 2000 });
```
Comment (styfle) [inline packages/now-cli/test/integration.js:770]: Why kill here?

File: packages/now-cli/src/util/dev/builder.ts (hunk 1 above)
Comment (lucleray) [reply packages/now-cli/src/util/dev/builder.ts:69]: Agreed, I changed the order 👍

**Review by styfle — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**